### PR TITLE
mgr/dashboard: display real health in rbd mirroring pools

### PR DESCRIPTION
**Pull request by pereman2**
One way mirroring in rbd was displaying rbd pool in error state. The rbd cli tool in contrast, was displaying it in unknown state.

This prs trys to replicate the behaviour of the cli as much as possible (reference: `src/tools/rbd/action/MirrorPool.cc::execute_status`).

Fixes: https://tracker.ceph.com/issues/57615
Signed-off-by: Pere Diaz Bou <pdiazbou@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by pereman2** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/controllers/rbd_mirroring.py [hunk 1]
```diff
@@ -113,45 +118,42 @@ def get_pools(daemons):  # pylint: disable=R0912, R0915
...
+    health_color = 'success'
+    if stat['health'] == MirrorHealth.MIRROR_HEALTH_ERROR:
+        health = 'Error'
+        health_color = 'error'
+    elif stat['health'] == MirrorHealth.MIRROR_HEALTH_WARNING:
+        health = 'Warning'
+        health_color = 'warning'
+    elif stat['health'] == MirrorHealth.MIRROR_HEALTH_UNKNOWN:
+        health = 'Unknown'
+        health_color = 'warning'
```
Comment (Pegonzal) [inline src/pybind/mgr/dashboard/controllers/rbd_mirroring.py:132]: Health unkown is now color 'warning' instead of previous color 'info'?

**Review by Pegonzal — COMMENTED** [review]
There are a couple tests failing from `test_rbd_mirroring.py` which seem related to the PR. Might need to take a look.

**Comment by pereman2** [comment]
jenkins test make check

**Comment by nizamial09** [comment]
@pereman2 test failures are related? and also is it possible to put some screenshots of the change (before and after)

**Comment by nizamial09** [comment]
jenkins test make check

**Review by nizamial09 — DISMISSED** [review]
[make-check ](https://jenkins.ceph.com/job/ceph-pull-requests/103844/consoleFull)failure

```
self = <dashboard.tests.test_rbd_mirroring.GetDaemonAndPoolsTest testMethod=test_get_pools_health>
mock_rbd = <MagicMock name='RBD' id='140098960839376'>

    @mock.patch('rbd.RBD')
    def test_get_pools_health(self, mock_rbd):
    
        mock_rbd_instance = mock_rbd.return_value
        mock_rbd_instance.mirror_peer_list.return_value = []
        test_cases = self._get_pool_test_cases()
        for new_status, mirror_mode, expected_output in test_cases:
            _status[1].update(new_status)
            daemon_status = {
                'json': json.dumps(_status)
            }
            mgr.get_daemon_status.return_value = daemon_status
            daemons = get_daemons()
            mock_rbd_instance.mirror_mode_get.return_value = mirror_mode
            res = get_pools(daemons)
            for k, v in expected_output.items():
>               self.assertTrue(v == res['rbd'][k])
E               AssertionError: False is not true

tests/test_rbd_mirroring.py:121: AssertionError
```

**Comment by nizamial09** [comment]
@pereman2 can you also link the tracker to the PR?

File: src/pybind/mgr/dashboard/controllers/rbd_mirroring.py (hunk 1 above)
Comment (pereman2) [reply src/pybind/mgr/dashboard/controllers/rbd_mirroring.py:132]: Because rbd mirror identifies Unkown as a Warning

**Comment by pereman2** [comment]
jenkins test make check

**Comment by pereman2** [comment]
I'm running the same test locally and it passes :/, let's see if retriggering helps

**Comment by pereman2** [comment]
jenkins test make check

**Comment by pereman2** [comment]
jenkins test make check

**Comment by Pegonzal** [comment]
> I'm running the same test locally and it passes :/, let's see if retriggering helps

@pereman2 I can see it happening locally as well:

`FAILED tests/test_rbd_mirroring.py::GetDaemonAndPoolsTest::test_get_pools_health - AssertionError: False is not true
FAILED tests/test_rbd_mirroring.py::RbdMirroringSummaryControllerTest::test_summary - AssertionError: {'warnings': 0, 'errors': 0} != {'errors': 0, 'warnings': 1}
`

**Review by avanthakkar — APPROVED** [review]
LGTM!

**Review by Pegonzal — APPROVED** [review]

**Pull request merged**